internal/util/observability/observability.go

```diff
@@ -111,3 +121,61 @@ func (ot *OtelTracer) Run(ctx context.Context) {
 
 	ot.l.Info("OTLP tracer stopped.")
 }
+
+// ExporterWithFilter is a span exporter that filters out spans with ExclusionAttribute and all their subspans.
+type ExporterWithFilter struct {
+	exporter otelsdktrace.SpanExporter
+}
+
+// ExclusionAttribute is the attribute that marks spans to be excluded.
+var ExclusionAttribute = attribute.KeyValue{Key: "excluded", Value: attribute.BoolValue(true)}
+
+// ExportSpans implements otelsdktrace.SpanExporter interface.
+func (e *ExporterWithFilter) ExportSpans(ctx context.Context, spans []otelsdktrace.ReadOnlySpan) error {
```

Comment:
Let me see if I understand what we are trying to do. Do we want to stop trace for envtool? I'm trying to understand the implementation of `ExclusionAttribute` to our usage, is it only for envtool? Currently, this is pretty tricky to understand.

Could a simpler solution be used? In https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#example-SpanProcessor-Filtered
there is `InstrumentationBlacklist` example which drops span with specified InstrumentationScope name.